app/views/users/show.html.erb

```diff
@@ -1,29 +1,14 @@
-<div class="user_container">
-        <div class="user_img">
-      <%= image_tag "profile-picture.jpg", size: "200x200" %>
-       </div>
-       <div class="user_data">
-        <h1><%= @user.name %></h1>
-        <p>Number of posts:<%= @user.posts_counter || 0 %></p></div>
-  </div>
-
-        <div class="user_bio">
-        <h2>Bio</h2>
-        <p><%= @user.bio %></p>
-        </div>
- <div class="comment_data">
-    <% @user.posts.each do |e| %>
-        <div class="post_container">
-            <h1>Post #<%= e.id %></h1>
-            <p><%= e.text %></p>
-            <span>Comments:<%= e.comments_counter || 0 %></span>
-            <span>Likes:<%= e.likes_counter || 0 %></span>
-            </div>
-       
-    <% end %>
-    </div>
-  <div class="button-wrapper">
-    <button class="button"><%=link_to "See all posts", user_posts_path(@user) %></button>
-  </div>
+<%= render partial: 'partials/user_partial', locals: { user: @user } %>
 
+<div >
+<p>Bio
+<p>
+<%= @user.bio%>
+</p>
+</p>
 
+<% @recent_post.each do |recent| %>
```

Comment:
- [ ] I am getting this `NoMethodError` when I click on a user to show user details. This Is because you don't have a `@recent_post` variable in your show action. Kindly add this variable in your show users' action. this is supposed to return the 3 most recent posts by that user.
```ruby
 def show
    @user = User.find(params[:id])
    @recent_post = @user.most_recent_post
  end
```